data/data-pipeline/data_pipeline/etl/score/tests/conftest.py

```diff
@@ -0,0 +1,118 @@
+import os
+from importlib import reload
+from pathlib import Path
+
+import pandas as pd
+import pytest
+from data_pipeline import config
+from data_pipeline.etl.score import etl_score_post, tests
+from data_pipeline.etl.score.etl_score_post import PostScoreETL
+
+
+def pytest_configure():
+    pytest.SNAPSHOT_DIR = Path(__file__).parent / "snapshots"
+
+
+@pytest.fixture(scope="session")
+def root(tmp_path_factory):
+    basetemp = Path.cwd() / "temp_dir"
+    os.environ["PYTEST_DEBUG_TEMPROOT"] = str(
+        basetemp
+    )  # this sets the location of the temp directory inside the project folder
+    basetemp.mkdir(parents=True, exist_ok=True)
+    root = tmp_path_factory.mktemp("root", numbered=False)
+    return root
+
+
+@pytest.fixture(autouse=True)
+def settings_override(monkeypatch, root):
+    reload(config)
+
+    monkeypatch.setattr(config.settings, "APP_ROOT", root)
+    return config.settings
+
+
+@pytest.fixture()
+def etl(monkeypatch, root):
+    reload(etl_score_post)
+
+    tmp_path = root / "tmp"
+    tmp_path.mkdir(parents=True, exist_ok=True)
+    etl = PostScoreETL()
+    monkeypatch.setattr(etl, "DATA_PATH", root)
+    monkeypatch.setattr(etl, "TMP_PATH", tmp_path)
+    return etl
+
+
+@pytest.fixture(scope="session")
+def sample_data_dir():
+    base_dir = Path(tests.__file__).resolve().parent
+    return base_dir / "sample_data"
+
+
+@pytest.fixture()
+def county_data_initial(sample_data_dir):
+    return sample_data_dir / "county_data_initial.csv"
+
+
+@pytest.fixture()
+def state_data_initial(sample_data_dir):
+    return sample_data_dir / "state_data_initial.csv"
+
+
+@pytest.fixture()
+def score_data_initial(sample_data_dir):
+    return sample_data_dir / "score_data_initial.csv"
+
+
+@pytest.fixture()
+def counties_transformed_expected():
+    return pd.DataFrame.from_dict(
+        data={
+            "State Abbreviation": pd.Series(["AL", "AL"], dtype="string"),
+            "GEOID": pd.Series(["01001", "01003"], dtype="string"),
+            "County Name": pd.Series(
+                ["AutaugaCounty", "BaldwinCounty"], dtype="object"
+            ),
+        },
+    )
+
+
+@pytest.fixture()
+def states_transformed_expected():
+    return pd.DataFrame.from_dict(
+        data={
+            "State Code": pd.Series(["01", "02", "04"], dtype="string"),
+            "State Name": pd.Series(["Alabama", "Alaska", "Arizona"], dtype="object"),
+            "State Abbreviation": pd.Series(["AL", "AK", "AZ"], dtype="string"),
+        },
+    )
+
+
+@pytest.fixture()
+def score_transformed_expected():
+    return pd.read_pickle(pytest.SNAPSHOT_DIR / "score_transformed_expected.pkl")
```

Comment:
I'm not sure about using pickle. It doesn't seem to me that we have files/fixtures big enough to warrant translating to/from binary, and it obfuscates the expected output such that if you wanted to troubleshoot you'd have to take an extra step (or more?) to see what's going on. Do we get anything from this other than some (arguably slight) performance or memory gains?

Reply:
Learned more synchronously: We're using these to preserve types in the dataframe so that we can accurately compare those. There may be other ways to solve this problem that are clearer than pickles (eg loading from sqlite or using tiangolo/sqlmodel or using snapshots; see comment below).

Reply:
Sounds like we need pickles for now, so let's add some docs and a class method to create the pickles. Let's at least add to the README for this PR. We'll create a new issue for spiking on using pytest-snapshot.